hil-test/build.rs

```diff
@@ -37,5 +37,9 @@ fn main() -> Result<(), Box<dyn Error>> {
     // Define all necessary configuration symbols for the configured device:
     config.define_symbols();
 
+    if cfg!(feature = "esp-wifi") {
```

Comment:
We should really emit this as part of esp-wifi or esp-hal's build script, I'll create an issue for that

Reply:
I think I tried that but it was ignored for library crates - maybe something changed since then 🤔

Reply:
Hmm yeah you're right, I forgot about that - only works in the final crate build.rs, not libs. Sorry for the noise!